core/src/main/java/com/datastrato/gravitino/catalog/CatalogManager.java

```diff
@@ -211,24 +211,28 @@ public void close() {
   @Override
   public NameIdentifier[] listCatalogs(Namespace namespace) throws NoSuchMetalakeException {
     NameIdentifier metalakeIdent = NameIdentifier.of(namespace.levels());
+    checkMetalakeExists(metalakeIdent);
 
-    boolean metalakeExists;
     try {
-      metalakeExists = store.exists(metalakeIdent, EntityType.METALAKE);
-    } catch (IOException e) {
-      LOG.error("Failed to do storage operation", e);
-      throw new RuntimeException(e);
-    }
+      return store.list(namespace, CatalogEntity.class, EntityType.CATALOG).stream()
+          .map(entity -> NameIdentifier.of(namespace, entity.name()))
+          .toArray(NameIdentifier[]::new);
 
-    if (!metalakeExists) {
-      throw new NoSuchMetalakeException(METALAKE_DOES_NOT_EXIST_MSG, metalakeIdent);
+    } catch (IOException ioe) {
+      LOG.error("Failed to list catalogs in metalake {}", metalakeIdent, ioe);
+      throw new RuntimeException(ioe);
     }
+  }
+
+  @Override
+  public Catalog[] listCatalogsInfo(Namespace namespace) throws NoSuchMetalakeException {
+    NameIdentifier metalakeIdent = NameIdentifier.of(namespace.levels());
+    checkMetalakeExists(metalakeIdent);
 
     try {
       return store.list(namespace, CatalogEntity.class, EntityType.CATALOG).stream()
-          .map(entity -> NameIdentifier.of(namespace, entity.name()))
-          .toArray(NameIdentifier[]::new);
-
+          .map(entity -> createCatalogWrapper(entity).catalog)
```

Comment:
It seems too heavy to create a catalog wrapper to get catalog info, are catalog information are stored in entity store, no need to create a classloader to load specific catalog?

Reply:
`CatalogEntity` does not implement the interface `Catalog`, how to get `Catalog` from `CatalogEntity`?

BTW, if it's too heavy to create a catalog wrapper, then does the method `loadCatalog` have the same issue? Should we refactor `loadCatalog` either? Or should we consider this in another separate issue?

Reply:
`loadCatalog` mostly means that users will use the catalog, so the related classpath and others should be loaded. But for list catalog, seems like users only need some information stored in entity store, no need to load the catalog to use.

Reply:
Also you just loaded it into the memory without putting into the management, the catalog object cannot be managed, it will lead to the memory leakage.

Reply:
fixed,  please help review again. @FANNG1 @jerryshao